securesystemslib/keys.py

```diff
@@ -395,6 +404,70 @@ def generate_ed25519_key(scheme='ed25519'):
 
 
 
+def generate_spx_key(scheme='spx'):
+  """
+  <Purpose>
+    Generate public and private SPX keys, In addition, a keyid identifier
+    generated for the returned SPX object.  The object returned conforms to
+    'securesystemslib.formats.SPXKEY_SCHEMA' and has the form:
+
+    {'keytype': 'spx',
+     'scheme': 'spx',
+     'keyid': 'f30a0870d026980100c0573bd557394f8c1bbd6...',
+     'keyval': {'public': '9ccf3f02b17f82febf5dd3bab878b767d8408...',
+                'private': 'ab310eae0e229a0eceee3947b6e0205dfab3...'}}
+
+    >>> spx_key = generate_spx_key()
+    >>> securesystemslib.formats.SPXKEY_SCHEMA.matches(spx_key)
+    True
+
+  <Arguments>
+    scheme:
+      The signature scheme used by the generated SPX key.
+
+  <Exceptions>
+    None.
+
+  <Side Effects>
+    The SPX keys are generated by calling the SPX routines provided by 'pyspx'.
+
+  <Returns>
+    A dictionary containing the SPX keys and other identifying information.
+    Conforms to 'securesystemslib.formats.SPXKEY_SCHEMA'.
+  """
+
+  # Are the arguments properly formatted?  If not, raise an
+  # 'securesystemslib.exceptions.FormatError' exceptions.
+  securesystemslib.formats.SPX_SIG_SCHEMA.check_match(scheme)
+
+  # Begin building the SPX key dictionary.
+  spx_key = {}
+  keytype = 'spx'
```

Comment:
Do we need `keytype` and `scheme` separately? I see that `SPX_SIG_SCHEMA` is set to `OneOf` but only `spx` is provided. Is this just leaving it open for for future enhancement?

Reply:
Yes we do. Currently we only support one `spx` scheme, i.e. `SPHINCS+-SHAKE256`, but it's conceivable to later add others (see [sphincs.org](https://sphincs.org/)).

I probably would have opted for a more descriptive scheme name, but that was the choice of the original author of this PR, and I think it is okay.